src/crypto.rs

```diff
@@ -1003,14 +1107,63 @@ mod test {
         assert_eq!(decoded, pub_key);
     }
 
+    #[test]
+    fn de_ser_ed25519_public_key_with_keyid_hash_algo() {
+        let pub_key = PrivateKey::from_pkcs8(ED25519_1_PK8, SignatureScheme::Ed25519)
+            .unwrap()
+            .public()
+            .clone();
+        let pub_key = PublicKey::from_ed25519_with_keyid_hash_algorithms(
+            pub_key.as_bytes().to_vec(),
+            python_tuf_compatibility_keyid_hash_algorithms(),
+        )
+        .unwrap();
+        let original = json!({
+            "keytype": "ed25519",
+            "scheme": "ed25519",
+            "keyid_hash_algorithms": ["sha256", "sha512"],
+            "keyval": {
+                "public": HEXLOWER.encode(pub_key.as_bytes()),
+            }
+        });
+
+        let encoded: PublicKey = serde_json::from_value(original.clone()).unwrap();
+        let decoded = serde_json::to_value(&encoded).unwrap();
+
+        assert_eq!(original, decoded);
+    }
+
+    #[test]
+    fn de_ser_ed25519_public_key_without_keyid_hash_algo() {
+        let pub_key = PrivateKey::from_pkcs8(ED25519_1_PK8, SignatureScheme::Ed25519)
+            .unwrap()
+            .public()
+            .clone();
+        let pub_key =
+            PublicKey::from_ed25519_with_keyid_hash_algorithms(pub_key.as_bytes().to_vec(), None)
+                .unwrap();
+        let original = json!({
+            "keytype": "ed25519",
+            "scheme": "ed25519",
+            "keyval": {
+                "public": HEXLOWER.encode(pub_key.as_bytes()),
+            }
+        });
+
+        let encoded: PublicKey = serde_json::from_value(original.clone()).unwrap();
+        let decoded = serde_json::to_value(&encoded).unwrap();
+
+        assert_eq!(original, decoded);
+    }
+
     #[test]
     fn serde_signature() {
```

Comment:
This function is getting its keyid changed because it includes keyid_hash_algorithms. Can you add another version that doesn't have keyid_hash_algorithms and confirm the keyid doesn't change?